File: labtasks/SIPLinphone/app/src/main/java/ro/pub/cs/systems/eim/lab10/siplinphone/MainActivity.java
package ro.pub.cs.systems.eim.lab10.siplinphone;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.LinearLayout;
import android.widget.RelativeLayout;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import org.linphone.core.Core;
import org.linphone.core.Factory;

public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main_activity);

        Factory factory = Factory.instance();
        Core core = factory.createCore(null, null, this);

        TextView status = findViewById(R.id.registration_status);
        status.setText("linphone lib version: " + core.getVersion());

        findViewById(R.id.register).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                findViewById(R.id.register_layout).setVisibility(View.GONE);
                findViewById(R.id.call_layout).setVisibility(View.VISIBLE);
            }
        });

        findViewById(R.id.unregister).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                findViewById(R.id.register_layout).setVisibility(View.VISIBLE);
                findViewById(R.id.call_layout).setVisibility(View.GONE);
            }
        });
    }
}

File: labtasks/SIPLinphone/app/src/main/java/ro/pub/cs/systems/eim/lab10/siplinphone/MainActivity.kt

package ro.pub.cs.systems.eim.lab10.siplinphone

import android.widget.*
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import org.linphone.core.*

class MainActivity: AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // Check the app/build.gradle to see how to import the LibLinphone SDK !!!

        setContentView(R.layout.main_activity)
        val coreVersion = findViewById<TextView>(R.id.core_version)

        // Core is the main object of the SDK. You can't do much without it.
       // To create a Core, we need the instance of the Factory.
        val factory = Factory.instance()

        // Some configuration can be done before the Core is created, for example enable debug logs.
        factory.setDebugMode(true, "Hello Linphone")

        val core = factory.createCore(null, null, this)

        // Now we can start using the Core object
        coreVersion.text = core.version
    }
}
